clash-lib/src/Clash/Normalize/Transformations.hs

```diff
@@ -953,6 +953,11 @@ removeUnusedExpr _ e@(collectArgsTicks -> (Data dc, [_,Right aTy,Right nTy,_,Lef
            in  changed v
       _ -> return e
 
+removeUnusedExpr _ e@(Letrec binds body) =
+  case removeUnusedBinders binds body of
```

Comment:
The issue with adding it to `removedUnusedExpr` is that the transformation is currently used in a topdown traversal, while you want to use `removeUnusedBinders` in a bottomup traversal after all the constants have been propagated and e.g. bodies of let-expressions stop refencing their binders. i.e. I would change: https://github.com/clash-lang/clash-compiler/blob/250aea43ce6111cd55cdb83e1ad30761024c9bb0/clash-lib/src/Clash/Normalize/Strategy.hs#L120

to

```haskell
inlineNR = bottomupR (apply "deadcode"  deadCode >-> apply "inlineNonRep" inlineNonRep)
```

Reply:
Done! Albeit with a small twist :-)